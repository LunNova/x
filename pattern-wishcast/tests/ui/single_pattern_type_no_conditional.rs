// SPDX-FileCopyrightText: 2025 LunNova
//
// SPDX-License-Identifier: MIT

//! Test helpful error for single pattern type with no conditional variants

#![feature(never_type)]

use pattern_wishcast::pattern_wishcast;

pattern_wishcast! {
	enum Value is <P: PatternFields> = {
		HostValue { value: String },
		TupleValue { elements: Vec<Self> },
	};

	// Only one pattern type, and it includes all variants
	type MyValue = Value is HostValue(_) | TupleValue(_);
}

fn main() {}
