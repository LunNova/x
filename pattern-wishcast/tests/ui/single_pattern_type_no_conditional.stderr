error: Enum `Value` has only one pattern type `MyValue`. Since there are no conditional variants, you don't need pattern support. Remove `is <P: PatternFields>` from the enum declaration and use a simple type alias instead: `type MyValue = Value;`
  --> tests/ui/single_pattern_type_no_conditional.rs:11:1
   |
11 | / pattern_wishcast! {
12 | |     enum Value is <P: PatternFields> = {
13 | |         HostValue { value: String },
14 | |         TupleValue { elements: Vec<Self> },
...  |
18 | |     type MyValue = Value is HostValue(_) | TupleValue(_);
19 | | }
   | |_^
   |
   = note: this error originates in the macro `pattern_wishcast` (in Nightly builds, run with -Z macro-backtrace for more info)
